### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by jelling** [Lycoon/scriptio#59]
Hey @Lycoon, first thanks for making this project. I stumbled across it while thinking about making something similar with TipTap. 

Presumably, this is an open source project. But have you thought about what license you want to release it under? I haven't decided yet whether I'm going to use it - or parts of it - but having an included license (or even just a response here) would make me feel more comfortable that I'm using it properly.

**Comment by Lycoon** [comment]
Hi @jelling 😊

Well, I have thought about it recently and I'm still considering which license to apply. The reason why I made Scriptio "open-source" - or at least publicly available - is first for the sake of transparency for the users. In addition to that core value I'm planning to and hope to keep Scriptio **completely free** as long as possible without any "paywall" limited stuff or anything (which would be quite different from current most solutions on the market, right?). Of course nothing is free but I don't want to put this burden on people who cannot afford it. Therefore I had in mind a donation-based business plan but this is out of scope for this issue.

Without current license Scriptio remains de facto "All rights reserved", which is not a situation I appreciate. I would like to be as permissive as possible for enabling devs further contributing to open-source. What I'm mostly afraid of are the very few greedy devs who see in the project a ready-to-go money-grabbing tool. I feel like keeping the authority until finding a license that fits to the ambition of the project is the safest for now. However I feel like [CC BY-NC-SA 4.0](http://creativecommons.org/licenses/by-nc-sa/4.0) would match quite well even if it implies forbidding commercial use and enforcing same license.

I hope my answer is complete enough. I'm open to suggestions as these software legal considerations are still new to me. Feel free to comment!

**Comment by jelling** [comment]
I might be one of those greedy devs :) Just kidding, but for transparency, I'm the founder of [RebootTV,](https://www.reboottv.net/) which is a business, although we are not charging anyone yet.

Currently, we have a Plate/Slate.JS based screenwriting tool as part of the platform. But we eventually want to add collaboration features so we are looking at TipTap as the new cloud-persistence features make that much easier. I found your project after searching for "screenwriting" on their Discord.

That said, we won't use any of your code as I respect your position and your mission. Also, from the couple of days I've spent with TipTap, it has been really easy to get a basic screenwriting app going.

Re: licensing and the risk of someone taking your work and starting a functioning business, more than half the pain of running a business is getting customers. And in the case of screenwriting, there are already a ton of low-cost or free options out there already. So I wouldn't worry too much.

If you are accepting PRs, I refactored `EditorAndSidebar` down to 150 LoC to make it easier to understand for people new to the project. I didn't change any logic, mostly just hoisting functionality out into custom hooks. Cheers!
